semgrep-core/synthesizing/Unit_synthesizer.ml

```diff
@@ -174,28 +174,40 @@ let unittest =
         let code = Parse_generic.parse_program file in
         let r = Range.range_of_linecol_spec range file in
         Naming_AST.resolve lang code;
-        let check_pats (_, pat) =
+        let check_pats (str, pat) =
           try
             let pattern = Parse_generic.parse_pattern lang pat in
             let e_opt = Range_to_AST.expr_at_range r code in
                match e_opt with
                  | Some e ->
-                    let matches_with_env = Semgrep_generic.match_any_any pattern (A.E e) in
-                    (* Debugging note: uses pattern_to_string for convenience, but really should *)
-                    (* match the code in the given file at the given range *)
-                    pr2 (AST_generic.show_any (A.E e));
-                    pr2 (AST_generic.show_any (pattern));
-                    assert_bool (spf "pattern:|%s| should match |%s" pat (PPG.pattern_to_string lang (A.E e)))
-                    (matches_with_env <> [])
-                 | None -> failwith (spf "Couldn't find range %s in %s" range file)
+                    let matches_with_env = Semgrep_generic.match_any_any
+                          pattern (A.E e) in
+                    (* Debugging note: uses pattern_to_string for convenience,
+                     * but really should match the code in the given file at
+                     * the given range *)
+                    if matches_with_env = []
+                    then begin
```

Comment:
I'll probably want to put this as the error message and not print it out each time